rp2-pio/pio.go

```diff
@@ -263,6 +298,28 @@ type pioHW struct {
 	IRQ_INT           [2]irqINTHW             // 0x12C..0x140
 }
 
+// RP2350 Programmable IO block
+type pioHW2350 struct {
```

Comment:
It seems this PR could be **_massively_** reduced in complexity if pioHW was a type shared between rp2040 and rp2350 targets. This is possible taking the same approach done in tinygo machine repo to implement rp2350 logic. Please revert pioHW to it's initial state and add registers with a `rp2350ExtraReg`-like constant 

See:
- https://github.com/tinygo-org/tinygo/blob/3869f76887feef6c444308e7e1531b7cac1bbd10/src/machine/machine_rp2_2350.go#L15
- https://github.com/tinygo-org/tinygo/blob/3869f76887feef6c444308e7e1531b7cac1bbd10/src/machine/machine_rp2_2040.go#L16
- https://github.com/tinygo-org/tinygo/blob/3869f76887feef6c444308e7e1531b7cac1bbd10/src/machine/machine_rp2_gpio.go#L36

Reply:
This will let us have the same files for both rp2350 and rp2040. no need to duplicate code

Reply:
Ah, yep. Fixed